riot/riot.py

```diff
@@ -457,6 +460,41 @@ def __hash__(self):
         h.update(self.full_pkg_str.encode())
         return int(h.hexdigest(), 16)
 
+    @property
+    def requirements(self) -> str:
+        """Requirements for dependencies with pinned versions."""
+        # Transform full_pkg_str into requirements.in format
+        pkgs = "\n".join(self.full_pkg_str.replace("'", "").split(" "))
+        _dir = os.path.join(DEFAULT_RIOT_PATH, "requirements")
+        os.makedirs(_dir, exist_ok=True)
+        in_path = os.path.join(_dir, "{}.in".format(self.short_hash))
+        subprocess.check_output(
+            [self.py.path(), "-m", "pip", "install", "pip-tools"],
```

Comment:
do we just need to add `pip-tools` as a default requirement in `setup.py`?

Reply:
or does it have to be in each venv?

Reply:
it has to be in each venv because it matters which version of python was used to install it.